src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -1405,5 +1686,119 @@ .maxstack  1
+                // (10,5): error CS0106: The modifier 'override' is not valid for this item
+                //     override data int P2 = 2;
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "override data int P2 = 2;").WithArguments("override").WithLocation(10, 5),
+                // (11,19): error CS0236: A field initializer cannot reference the non-static field, method, or property 'C.P1'
+                //     data int P3 = P1;
+                Diagnostic(ErrorCode.ERR_FieldInitRefNonstatic, "P1").WithArguments("C.P1").WithLocation(11, 19),
+                // (12,22): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     data string P4 = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(12, 22)
+            );
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var props = tree.GetRoot().DescendantNodes().OfType<DataPropertyDeclarationSyntax>().ToList();
+            var p1 = (IPropertySymbol)model.GetDeclaredSymbol(props[0])!;
+            Assert.NotNull(p1);
+            Assert.Equal("P1", p1.Name);
+            Assert.False(p1.IsReadOnly);
+            Assert.False(p1.IsWriteOnly);
+            Assert.True(p1.SetMethod!.IsInitOnly);
+            Assert.False(p1.IsAbstract);
+            Assert.False(p1.IsOverride);
+            Assert.False(p1.IsVirtual);
+
+            var typeInfo = model.GetTypeInfo(props[0].Initializer!.Value);
+            Assert.Equal(SpecialType.System_Int32, typeInfo.Type!.SpecialType);
+            Assert.Equal(CodeAnalysis.NullableFlowState.NotNull, typeInfo.Nullability.FlowState);
+
+            var initializer = props[1].Initializer;
+            Assert.NotNull(initializer);
+            typeInfo = model.GetTypeInfo(initializer!.Value);
+            Assert.Equal(SpecialType.System_Int32, typeInfo.Type!.SpecialType);
+
+            var p2 = (IPropertySymbol)model.GetDeclaredSymbol(props[1])!;
+            Assert.NotNull(p2);
+            Assert.Equal("P2", p2.Name);
+            Assert.False(p2.IsReadOnly);
+            Assert.False(p2.IsWriteOnly);
+            Assert.True(p2.SetMethod!.IsInitOnly);
+            Assert.False(p2.IsAbstract);
+            Assert.False(p2.IsOverride);
+            Assert.False(p2.IsVirtual);
+
+            var symbolInfo = model.GetSymbolInfo(props[2].Initializer!.Value);
+            Assert.True(symbolInfo.Symbol!.Equals(p1, SymbolEqualityComparer.Default));
+
+            var p4 = (IPropertySymbol)model.GetDeclaredSymbol(props[3])!;
+            Assert.Equal(CodeAnalysis.NullableAnnotation.NotAnnotated, p4.Type.NullableAnnotation);
+            typeInfo = model.GetTypeInfo(props[3].Initializer!.Value);
+            Assert.Equal(CodeAnalysis.NullableFlowState.MaybeNull, typeInfo.Nullability.FlowState);
+        }
```

Comment:
Please also add a test verifying symbols from PE.

Reply:
That's what the Attribute test does. This is only testing the semantic model.